src/Util/Profiler.php

```diff
@@ -0,0 +1,235 @@
+<?php
+
+namespace Friendica\Util;
+
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\ContainerInterface;
+use Psr\Container\NotFoundExceptionInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * A class to store profiling data
+ * It can handle different logging data for specific functions or global performance measures
+ *
+ * It stores the data as log entries (@see LoggerInterface )
+ */
+class Profiler implements ContainerInterface
+{
+	/**
+	 * @var array The global performance array
+	 */
+	private $performance;
+	/**
+	 * @var array The function specific callstack
+	 */
+	private $callstack;
+	/**
+	 * @var bool True, if the Profiler is enabled
+	 */
+	private $enabled;
+	/**
+	 * @var bool True, if the Profiler should measure the whole rendertime including functions
+	 */
+	private $rendertime;
+
+	/**
+	 * @var LoggerInterface The profiler logger
+	 */
+	private $logger;
+
+	/**
+	 * @param LoggerInterface $logger The profiler logger
+	 * @param bool $enabled           True, if the Profiler is enabled
+	 * @param bool $renderTime        True, if the Profiler should measure the whole rendertime including functions
+	 */
+	public function __construct(LoggerInterface $logger, $enabled = false, $renderTime = false)
+	{
+		$this->enabled = $enabled;
+		$this->rendertime = $renderTime;
+		$this->logger = $logger;
+		$this->reset();
+	}
+
+	/**
+	 * Saves a timestamp for a value - f.e. a call
+	 * Necessary for profiling Friendica
+	 *
+	 * @param int $timestamp the Timestamp
+	 * @param string $value A value to profile
+	 * @param string $callstack The callstack of the current profiling data
+	 */
+	public function saveTimestamp($timestamp, $value, $callstack = '')
+	{
+		if (!$this->enabled) {
+			return;
+		}
+
+		$duration = (float) (microtime(true) - $timestamp);
+
+		if (!isset($this->performance[$value])) {
+			// Prevent ugly E_NOTICE
+			$this->performance[$value] = 0;
+		}
+
+		$this->performance[$value] += (float) $duration;
+		$this->performance['marktime'] += (float) $duration;
+
+		if (!isset($this->callstack[$value][$callstack])) {
+			// Prevent ugly E_NOTICE
+			$this->callstack[$value][$callstack] = 0;
+		}
+
+		$this->callstack[$value][$callstack] += (float) $duration;
+	}
+
+	/**
+	 * Resets the performance and callstack profiling
+	 *
+	 * @param bool $performance If true, reset the performance (Default true)
+	 * @param bool $callstack   If true, reset the callstack (Default true)
+	 */
+	public function reset($performance = true, $callstack = true)
```

Comment:
Please refactor this function into two separate `resetPerformance` and `resetCallstack` methods who will be called separately by the calling code if any parameter was `false` in the current state.

You can keep this method that will take no parameters and just call both `resetPerformance` and `resetCallstack`.

Reply:
done.

Reply:
My goal was to get rid the two boolean parameters. Please adapt calls to `Profiler->reset` accordingly.